### PR TITLE
[AMS-514] Handling non US-ASCII symbols in HTTP headers according to RFC-5987

### PR DESCRIPTION
**Pull request by denisivan0v**

### CODE REVIEW

File: tests/VStore.UnitTests/MetadataCollectionWrapperTests.cs [hunk 1]
```diff
@@ -22,23 +22,34 @@ public void ShouldWrapUsingInvariantCulture()
...
 
-            var unwrapped = wrapper.Unwrap();
-            var keys = new string[unwrapped.Keys.Count];
-            unwrapped.Keys.CopyTo(keys, 0);
-            Assert.True(IsBase64String(unwrapped[keys[0]]));
+        [Fact]
+        public void ShouldWorkForNonUsAsciiSymbols()
+        {
+            var wrapper = MetadataCollectionWrapper.For(new MetadataCollection());
+            var value = "Вася Пупкин";
```
Comment (xakep139) [inline tests/VStore.UnitTests/MetadataCollectionWrapperTests.cs]: Good naming!

File: src/VStore/S3/MetadataCollectionWrapper.cs [hunk 2]
```diff
@@ -8,51 +7,79 @@ namespace NuClear.VStore.S3
...
+            {
+                headerName += "*";
+            }
+
+            return headerName;
+        }
+
+        public static bool IsJson(string input)
+        {
+            input = input.Trim();
```
Comment (xakep139) [inline src/VStore/S3/MetadataCollectionWrapper.cs]: It's better to create new variable

File: src/VStore/Sessions/SessionManagementService.cs [hunk 3]
```diff
@@ -236,8 +236,8 @@ public sealed class SessionManagementService
                             getResponse.ResponseStream);
                     }
 
-                    var metadataWrapper = MetadataCollectionWrapper.For(getResponse.Metadata);
-                    var fileName = metadataWrapper.Read<string>(MetadataElement.Filename);
+                var metadataWrapper = MetadataCollectionWrapper.For(getResponse.Metadata);
+                var fileName = metadataWrapper.ReadEncoded<string>(MetadataElement.Filename);
```
Comment (xakep139) [inline src/VStore/Sessions/SessionManagementService.cs]: Spacing

File: src/VStore/Sessions/SessionStorageReader.cs [hunk 4]
```diff
@@ -41,9 +41,9 @@ public async Task<(SessionDescriptor SessionDescriptor, AuthorInfo AuthorInfo, D
                         throw new SessionExpiredException(sessionId, expiresAt);
                     }
 
-                    var author = metadataWrapper.Read<string>(MetadataElement.Author);
-                    var authorLogin = metadataWrapper.Read<string>(MetadataElement.AuthorLogin);
-                    var authorName = metadataWrapper.Read<string>(MetadataElement.AuthorName);
+            var author = metadataWrapper.Read<string>(MetadataElement.Author);
+            var authorLogin = metadataWrapper.Read<string>(MetadataElement.AuthorLogin);
+            var authorName = metadataWrapper.ReadEncoded<string>(MetadataElement.AuthorName);
```
Comment (xakep139) [inline src/VStore/Sessions/SessionStorageReader.cs]: Spacing

File: tests/VStore.UnitTests/MetadataCollectionWrapperTests.cs [hunk 5]
```diff
@@ -22,23 +22,44 @@ public void ShouldWrapUsingInvariantCulture()
...
+            var encodedValue = Uri.EscapeDataString(value);
+            wrapper.WriteEncoded(MetadataElement.AuthorName, "utf-8''" + encodedValue);
+            var res = wrapper.ReadEncoded<string>(MetadataElement.AuthorName);
+            Assert.Equal(value, res);
         }
 
-        private bool IsBase64String(string s)
+        [Fact]
+        public void ShouldWorkForAlmostJson()
```
Comment (xakep139) [inline tests/VStore.UnitTests/MetadataCollectionWrapperTests.cs]: I've added the test to show failures

Comment (denisivan0v) [inline tests/VStore.UnitTests/MetadataCollectionWrapperTests.cs]: Fixed

File: src/VStore/Sessions/SessionStorageReader.cs (hunk 4 above)
Comment (denisivan0v) [inline src/VStore/Sessions/SessionStorageReader.cs]: Fixed in db428aa

File: src/VStore/Sessions/SessionManagementService.cs (hunk 3 above)
Comment (denisivan0v) [inline src/VStore/Sessions/SessionManagementService.cs]: FixFixed in db428aa

File: src/VStore/S3/MetadataCollectionWrapper.cs (hunk 2 above)
Comment (denisivan0v) [inline src/VStore/S3/MetadataCollectionWrapper.cs]: Method removed

**Comment by denisivan0v** [comment]
@xakep139 take a look at 1645a04

**Pull request merged**